obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+    """
+    with open(sfile, 'r') as f:
+        spec_inf = _read_spectral_info(f=f)
+    return spec_inf
+
+
+def _read_spectral_info(f):
+    """Internal spectral reader.
+
+    :type f: file
+    :param f: File open in read mode.
+
+    :returns: dictionary of spectral information, units as in seisan manual, \
+        expect for logs which have been converted to floats.
+    """
+    event = _readheader(f=f)
+    f.seek(0)
+    origin_date = UTCDateTime(event.origins[0].time.date)
+    relevant_lines = []
+    for line in f:
+        if str(line[1:5]) == str('SPEC'):
+            relevant_lines.append(line)
+    spec_inf = []
+    if len(relevant_lines) == 0:
+        return spec_inf
+    for line in relevant_lines:
+        spec_str = str(line.rstrip().lstrip())
+        if str(spec_str[5:12]) == str('AVERAGE'):
+            info = {}
+            info['station'] = str('AVERAGE')
+            info['channel'] = str('')
+            info['moment'] = _float_conv(spec_str[16:22])
+            if info['moment'] == 999:
+                info['moment'] = np.nan
+            else:
+                info['moment'] = 10 ** info['moment']
+            info['stress_drop'] = _float_conv(spec_str[24:30])
+            info['spectral_level'] = _float_conv(spec_str[32:38])
+            if info['spectral_level'] == 999:
+                info['spectral_level'] = np.nan
+            else:
+                info['spectral_level'] = 10 ** info['spectral_level']
+            info['corner_freq'] = _float_conv(spec_str[40:46])
+            info['source_radius'] = _float_conv(spec_str[47:54])
+            info['decay'] = _float_conv(spec_str[56:62])
+            info['window_length'] = _float_conv(spec_str[64:70])
+            info['moment_mag'] = _float_conv(spec_str[72:78])
+            spec_inf.append(info)
+            continue
+        if str(spec_str[5:7]) == str('SD'):
+            info = {}
+            info['station'] = str('STANDARD_DEVIATION')
+            info['channel'] = str('')
+            info['moment'] = _float_conv(spec_str[16:22])
+            if info['moment'] == 999:
+                info['moment'] = np.nan
+            else:
+                info['moment'] = 10 ** info['moment']
+            info['stress_drop'] = _float_conv(spec_str[24:30])
+            info['spectral_level'] = _float_conv(spec_str[32:38])
+            if info['spectral_level'] == 999:
+                info['spectral_level'] = np.nan
+            else:
+                info['spectral_level'] = 10 ** info['spectral_level']
+            info['corner_freq'] = _float_conv(spec_str[40:46])
+            info['source_radius'] = _float_conv(spec_str[47:54])
+            info['decay'] = _float_conv(spec_str[56:62])
+            info['window_length'] = _float_conv(spec_str[64:70])
+            info['moment_mag'] = _float_conv(spec_str[72:78])
+            spec_inf.append(info)
+            continue
+        stachans = [str('.'.join([i['station'], i['channel']]))
```

Comment:
Actually, if you're going to go find the station/channel again (which you do on line 445), then `spec_inf` should be a dictionary, keyed by `(station, channel)`, which is basically a `set`+`list`.